### ISSUE TITLE
Can't build REE

### ISSUE THREAD

**Issue by brauliobo** [sstephenson/ruby-build#758]
```
~/V/Animais ❯❯❯ rbenv install ree-1.8.7-2012.02
grep: aviso: a variável de ambiente GREP_OPTIONS está obsoleta; use um alias ou script
Downloading ruby-enterprise-1.8.7-2012.02.tar.gz...
-> http://dqw8nmjcqpjn7.cloudfront.net/ecf4a6d4c96b547b3bf4b6be14e082ddaa781e83ad7f69437cd3169fb7576e42
Installing ruby-enterprise-1.8.7-2012.02...

BUILD FAILED (Arch rolling using ruby-build 20150303)

Inspect or clean up the working tree at /tmp/braulio/ruby-build.20150511174619.3606
Results logged to /tmp/braulio/ruby-build.20150511174619.3606.log

Last 10 log lines:
src/tcmalloc.cc:1672:54: error: conflicting declaration ‘void* (* __memalign_hook)(size_t, size_t, const void*)’
 void *(*__memalign_hook)(size_t, size_t, const void *) = MemalignOverride;
                                                      ^
In file included from src/tcmalloc.cc:101:0:
/usr/include/malloc.h:160:39: note: previous declaration as ‘void* (* volatile __memalign_hook)(size_t, size_t, const void*)’
 extern void *(*__MALLOC_HOOK_VOLATILE __memalign_hook)(size_t __alignment,
                                       ^
Makefile:3317: recipe for target 'libtcmalloc_minimal_la-tcmalloc.lo' failed
make: *** [libtcmalloc_minimal_la-tcmalloc.lo] Error 1
grep: aviso: a variável de ambiente GREP_OPTIONS está obsoleta; use um alias ou script
```

**Comment by hsbt** [comment]
What's your OS?

**Comment by brauliobo** [comment]
archlinux

**Comment by hsbt** [comment]
@brauliobo It's not ruby-build issue. If you want to build REE on ArchLinux, You need to report upstream as REE developer or fixed issue with yourself.

We do not support old ruby with modern OS.

**Comment by mislav** [comment]
@brauliobo I'm sorry, but it seems that REE is incompatible with Arch. REE isn't developed for 3 years now, so it's not surprising that it doesn't build on newer OS. You will need to either google for workarounds how to build on Arch (maybe someone has a patch you can apply with `--patch` option, or maybe there's a magic environment variable), but that's not something we can help you with.

Closing as there's little we can do about this from ruby-build.

**Comment by brauliobo** [comment]
@mislav I don't this might be a problem with a newer system, but RVM could successfully build REE (see output below), so I guess it succeds because one or more of the patches RVM applies.
```
~/P/n/cirandas git:cirandas ❯❯❯ rvm reinstall ree                                                                                                                   ✭ ◼
ree-1.8.7-2012.02 - #removing src/ree-1.8.7-2012.02 - please wait
ree-1.8.7-2012.02 - #removing rubies/ree-1.8.7-2012.02 - please wait
Searching for binary rubies, this might take some time.
No binary rubies available for: arch/libc-2.21/x86_64/ree-1.8.7-2012.02.
It is not possible to build movable binaries for rubies 1.8-1.9.2, but you can do it for your system only.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.

Warning! Requested ruby installation which requires another ruby available - installing ruby-1.8.7-p374 first.

Searching for binary rubies, this might take some time.
No binary rubies available for: arch/libc-2.21/x86_64/ruby-1.8.7-p374.
It is not possible to build movable binaries for rubies 1.8-1.9.2, but you can do it for your system only.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for arch.
Requirements installation successful.
grep: aviso: a variável de ambiente GREP_OPTIONS está obsoleta; use um alias ou script
Installing Ruby from source to: /home/braulio/.rvm/rubies/ruby-1.8.7-p374, this may take a while depending on your cpu(s)...
ruby-1.8.7-p374 - #downloading ruby-1.8.7-p374, this may take a while depending on your connection...
ruby-1.8.7-p374 - #extracting ruby-1.8.7-p374 to /home/braulio/.rvm/src/ruby-1.8.7-p374 - please wait
ruby-1.8.7-p374 - #applying patch /home/braulio/.rvm/patches/ruby/1.8.7/stdout-rouge-fix.patch - please wait
ruby-1.8.7-p374 - #applying patch /home/braulio/.rvm/patches/ruby/1.8.7/no_sslv2.diff - please wait
ruby-1.8.7-p374 - #applying patch /home/braulio/.rvm/patches/ruby/GH-488.patch - please wait
ruby-1.8.7-p374 - #applying patch /home/braulio/.rvm/patches/ruby/ssl_no_ec2m.patch - please wait
ruby-1.8.7-p374 - #configuring - please wait
ruby-1.8.7-p374 - #post-configuration - please wait
ruby-1.8.7-p374 - #compiling - please wait
ruby-1.8.7-p374 - #installing - please wait
ruby-1.8.7-p374 - #making binaries executable - please wait
ruby-1.8.7-p374 - #downloading rubygems-2.0.16
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  332k  100  332k    0     0  54484      0  0:00:06  0:00:06 --:--:-- 79470
No checksum for downloaded archive, recording checksum in user configuration.
ruby-1.8.7-p374 - #extracting rubygems-2.0.16 - please wait
ruby-1.8.7-p374 - #removing old rubygems - please wait
ruby-1.8.7-p374 - #installing rubygems-2.0.16 - please wait
ruby-1.8.7-p374 - #gemset created /home/braulio/.rvm/gems/ruby-1.8.7-p374@global
ruby-1.8.7-p374 - #importing gemset /home/braulio/.rvm/gemsets/ruby/1.8.7/global.gems - please wait
ruby-1.8.7-p374 - #importing gemset /home/braulio/.rvm/gemsets/ruby/1.8.7/global.gems - please wait
ruby-1.8.7-p374 - #generating global wrappers - please wait
ruby-1.8.7-p374 - #gemset created /home/braulio/.rvm/gems/ruby-1.8.7-p374
ruby-1.8.7-p374 - #importing gemsetfile /home/braulio/.rvm/gemsets/default.gems evaluated to empty gem list
ruby-1.8.7-p374 - #generating default wrappers - please wait
ruby-1.8.7-p374 - #adjusting #shebangs for (gem irb erb ri rdoc testrb rake).
Install of ruby-1.8.7-p374 - #complete 
WARNING: Please be aware that you just installed a ruby that is no longer maintained (2014-02-23), for a list of maintained rubies visit:

    http://bugs.ruby-lang.org/projects/ruby/wiki/ReleaseEngineering

Please consider upgrading to ruby-2.2.0 which will have all of the latest security patches.
Ruby was built without documentation, to build it run: rvm docs generate-ri
Installing Ruby Enterprise Edition from source to: /home/braulio/.rvm/rubies/ree-1.8.7-2012.02
ree-1.8.7-2012.02 - #fetching (ruby-enterprise-1.8.7-2012.02)
ree-1.8.7-2012.02 - #extracting ruby-enterprise-1.8.7-2012.02 to /home/braulio/.rvm/src/ree-1.8.7-2012.02 - please wait
Checking requirements for arch.
Requirements installation successful.
grep: aviso: a variável de ambiente GREP_OPTIONS está obsoleta; use um alias ou script
ree-1.8.7-2012.02 - #applying patch /home/braulio/.rvm/patches/ree/1.8.7/tcmalloc.patch - please wait
ree-1.8.7-2012.02 - #applying patch /home/braulio/.rvm/patches/ree/1.8.7/stdout-rouge-fix.patch - please wait
ree-1.8.7-2012.02 - #applying patch /home/braulio/.rvm/patches/ree/1.8.7/no_sslv2.diff - please wait
ree-1.8.7-2012.02 - #applying patch /home/braulio/.rvm/patches/ree/ssl_no_ec2m.patch - please wait
ree-1.8.7-2012.02 - #applying patch /home/braulio/.rvm/patches/ree/1.8.7/p358-p374.patch - please wait
ree-1.8.7-2012.02 - #applying patch /home/braulio/.rvm/patches/ree/GH-488.patch - please wait
ree-1.8.7-2012.02 - #applying patch /home/braulio/.rvm/patches/ree/lib64.patch - please wait
ree-1.8.7-2012.02 - #applying patch /home/braulio/.rvm/patches/ree/ruby-binary-shebang.patch - please wait
ree-1.8.7-2012.02 - #installing - please wait
ree-1.8.7-2012.02 - #installing - please wait
ree-1.8.7-2012.02 - #making binaries executable - please wait
ree-1.8.7-2012.02 - #downloading rubygems-2.0.16
ree-1.8.7-2012.02 - #extracting rubygems-2.0.16 - please wait
ree-1.8.7-2012.02 - #removing old rubygems - please wait
ree-1.8.7-2012.02 - #installing rubygems-2.0.16 - please wait
ree-1.8.7-2012.02 - #gemset created /home/braulio/.rvm/gems/ree-1.8.7-2012.02@global
ree-1.8.7-2012.02 - #importing gemset /home/braulio/.rvm/gemsets/global.gems - please wait
...
```

**Comment by mislav** [comment]
Yes, RVM generally does a better job at maintaining patches for various cross-platform compilation problems. Glad you worked around your problem!

Other users of ruby-build stumbling on the same problem can always go fetch these patches from the RVM project

```
patches/ree/1.8.7/tcmalloc.patch
patches/ree/1.8.7/stdout-rouge-fix.patch
patches/ree/1.8.7/no_sslv2.diff
patches/ree/ssl_no_ec2m.patch
patches/ree/1.8.7/p358-p374.patch
patches/ree/GH-488.patch
patches/ree/lib64.patch
patches/ree/ruby-binary-shebang.patch
```

and combine them in a single patch to ruby-build like so:

```
cat ree-patches/* | rbenv install --patch ree-1.8.7-2012.02
```
